### PR TITLE
[Macro] Switching micromolecules mode to macromolecules(readonly)

### LINKED ISSUE #3330: Macro: Switching micromolecules mode to macromolecules(readonly)
**Background**
Users want to switch between macro/micro modes to clearly see the structures on both levels.

**Solution**
This task is the first iteration. It includes only readonly mode of structures which were created in different mode. All entities should keep their positions on canvas in both modes.

Macromolecules in micromode will be represented as SGroups with bonds and ability to see preview of monomer structure.
image

Micromolecules in macromode will be represented as CHEMs with generated name(F1, F2, ...Fn).

### PR DESCRIPTION
**Pull request by rrodionov91**
Closes #3330 

## How the feature works? / How did you fix the issue?
(Screenshots, videos, or GIFs, if applicable)

## Check list
- [ ] unit-tests written
- [ ] e2e-tests written
- [ ] documentation updated
- [ ] PR name follows the pattern `#1234 – issue name`
- [ ] branch name doesn't contain '#'
- [ ] PR is linked with the issue
- [ ] base branch (master or release/xx) is correct
- [ ] task status changed to "Code review"
- [ ] reviewers are notified about the pull request

### CODE REVIEW

File: packages/ketcher-core/src/application/render/restruct/rebond.ts [hunk 1]
```diff
@@ -59,8 +60,18 @@ class ReBond extends ReObject {
 
   static bondRecalc(bond: ReBond, restruct: ReStruct, options: any): void {
     const render = restruct.render;
-    const atom1 = restruct.atoms.get(bond.b.begin);
-    const atom2 = restruct.atoms.get(bond.b.end);
+    const sgroup1 = restruct.molecule.getGroupFromAtomId(bond.b.begin);
+    const sgroup2 = restruct.molecule.getGroupFromAtomId(bond.b.end);
+    const atom1 = restruct.atoms.get(
```
Comment (StarlaStarla) [inline packages/ketcher-core/src/application/render/restruct/rebond.ts:65]: Could we rename the atom1 to atomStart maybe? alos the others

**Review by Nitvex — CHANGES_REQUESTED** [review]

File: packages/ketcher-core/src/application/editor/Editor.ts [hunk 2]
```diff
@@ -214,4 +249,297 @@ export class CoreEditor {
 
     return false;
   }
+
+  private static convertMonomerToMonomerMicromolecule(
```
[the remark this answers is not in the document]
Comment (rrodionov91) [reply packages/ketcher-core/src/application/editor/Editor.ts:253]: It is not just sgroup. It is class MonomerMicromolecule which is extended from sgroup with overrides of some methods. And we use it to convert it back to specific monomers in macromolecules mode.

File: packages/ketcher-core/src/domain/serializers/ket/ketSerializer.ts [hunk 3]
```diff
@@ -204,118 +213,298 @@ export class KetSerializer implements Serializer<Struct> {
...
-          const template = parsedFileContent.root.templates.find(
-            (template) => template.id === node.templateId,
-          );
+          const template = parsedFileContent[
+            setMonomerTemplatePrefix(nodeDefinition.templateId)
+          ] as IKetMonomerTemplate;
           assert(template);
-          const struct = this.fillStruct(template);
-          command.merge(monomerToDrawingEntity(node, template, struct));
+          const struct = this.fillStruct({
```
[the remark this answers is not in the document]
Comment (rrodionov91) [reply packages/ketcher-core/src/domain/serializers/ket/ketSerializer.ts:299]: To be able to use fillStruct method which is micromolecules deserializing method. It works with such input:
```
{
  root: {
    nodes: [{ $ref: 'mol0' }],
  },
  mol0: {
    type: 'molecule',
    ...atoms, bonds, e.t.c
  },
}
```
So for macromolecules in ket file for each monomer we have also a template which structure is exactly same as this part with atoms, bonds, e.t.c above.

File: packages/ketcher-core/src/application/render/restruct/rebond.ts (hunk 1 above)
Comment (rrodionov91) [reply packages/ketcher-core/src/application/render/restruct/rebond.ts:65]: done

File: packages/ketcher-core/src/application/editor/Editor.ts [hunk 4]
```diff
@@ -214,4 +249,297 @@ export class CoreEditor {
...
+
+    monomerMicromolecule.data.name = monomer.monomerItem.label;
+    monomerMicromolecule.data.expanded = false;
+    monomerMicromolecule.id = sgroupId;
+    monomerMicromolecule.pp = monomer.position;
+
+    return monomerMicromolecule;
+  }
+
+  private static addMonomerAtomToStruct(
```
[the remark this answers is not in the document]
Comment (rrodionov91) [reply packages/ketcher-core/src/application/editor/Editor.ts:271]: done

File: packages/ketcher-core/src/application/editor/Editor.ts [hunk 5]
```diff
@@ -214,4 +249,297 @@ export class CoreEditor {
...
+    return { atomId, atom: atomClone };
+  }
+
+  private static findAttachmentPointAtom(
+    sgroup: SGroup,
+    polymerBond: PolymerBond,
+    monomer: BaseMonomer,
+    struct: Struct,
+  ) {
+    if (
```
[the remark this answers is not in the document]
Comment (rrodionov91) [reply packages/ketcher-core/src/application/editor/Editor.ts]: Done

File: packages/ketcher-core/src/application/render/renderers/BaseMonomerRenderer.ts [hunk 6]
```diff
@@ -224,9 +236,20 @@ export abstract class BaseMonomerRenderer extends BaseRenderer {
...
+    const monomerSymbolBoundingClientRect =
+      this.monomerSymbolBoundingClientRect;
     // we need to convert monomer coordinates(stored in angstroms) to pixels.
     // it needs to be done in view layer of application (like renderers)
-    return Scale.modelToCanvas(this.monomer.position, this.editorSettings);
+    const monomerPositionInPixels = Scale.modelToCanvas(
+      this.monomer.position,
+      this.editorSettings,
+    );
+    console.log(this.monomer.position, monomerPositionInPixels);
```
[the remark this answers is not in the document]
Comment (rrodionov91) [reply packages/ketcher-core/src/application/render/renderers/BaseMonomerRenderer.ts]: Done

File: packages/ketcher-core/src/domain/entities/BaseMonomer.ts [hunk 7]
```diff
@@ -221,13 +223,94 @@ export class BaseMonomer extends DrawingEntity {
...
+      side: (attachmentPointNumber) =>
+        attachmentPointNumber +
+        Number(!('R1' in attachmentPointDictionnary)) -
+        Number(!('R2' in attachmentPointDictionnary)),
+    };
+    this.monomerItem.attachmentPoints.forEach(
+      (attachmentPoint, attachmentPointIndex) => {
+        const attachmentPointNumber = attachmentPointIndex + 1;
+        attachmentPointDictionnary[
+          attachmentPoint.label ||
```
[the remark this answers is not in the document]
Comment (rrodionov91) [reply packages/ketcher-core/src/domain/entities/BaseMonomer.ts:250]: Done

File: packages/ketcher-core/src/domain/serializers/ket/ketSerializer.ts [hunk 8]
```diff
@@ -204,118 +213,298 @@ export class KetSerializer implements Serializer<Struct> {
...
+  ) {
+    CoreEditor.convertStructToDrawingEntities(struct, drawingEntitiesManager);
+
+    const { serializedMacromolecules, micromoleculesStruct } =
+      this.serializeMacromolecules(new Struct(), drawingEntitiesManager);
+
+    const serializedMicromoleculesStruct = JSON.parse(
+      this.serializeMicromolecules(micromoleculesStruct),
+    );
+    console.log(serializedMicromoleculesStruct, serializedMacromolecules);
```
[the remark this answers is not in the document]
Comment (rrodionov91) [reply packages/ketcher-core/src/domain/serializers/ket/ketSerializer.ts]: Done

**Review by Nitvex — APPROVED** [review]

**Pull request merged**